### ISSUE TITLE
Repeating Error while Sync: "Python int too large to convert to SQLite INTEGER"

### ISSUE THREAD

**Issue by stefan817** [samschott/maestral#609]
While syncing I get a repeating error "Python int too large to convert to SQLite INTEGER" and `maestral status` shows the message
```
Account:      xxxxxxx@xxx.xxx (Pro)
Usage:        39.1% of 2.2 TB used
Status:       Paused
Sync errors:  0

An unexpected error occurred
Please restart Maestral to continue syncing.
```
It mostly happens, after there was just one files downloaded. 
After stopping and starting the daemon again, the syncing goes on for some seconds or maybe a minute. Then the same again. 
It seems as if the syncing is successful, but it takes weeks to just sync one folder (others are excluded). 

From the log:
```
2022-02-15 15:02:39 manager INFO: Connected
2022-02-15 15:02:44 sync INFO: Resuming indexing
2022-02-15 15:02:46 sync INFO: Indexing 449994...
2022-02-15 15:02:46 sync INFO: Syncing ↓ 1/500
2022-02-15 15:02:48 sync INFO: Syncing ↓ 151/500
2022-02-15 15:02:50 sync INFO: Syncing ↓ 297/500
2022-02-15 15:02:56 manager ERROR: Unexpected error
Traceback (most recent call last):
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/manager.py", line 806, in _handle_sync_thread_errors
    yield
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/manager.py", line 784, in startup_worker
    self.sync.download_sync_cycle(client)
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/sync.py", line 2999, in download_sync_cycle
    downloaded = self.apply_remote_changes(changes)
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/sync.py", line 3167, in apply_remote_changes
    for n, r in enumerate(res):
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 586, in result_iterator
    yield fs.pop().result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/sync.py", line 3506, in _create_local_entry
    res = self._on_remote_file(event, client)
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/sync.py", line 3665, in _on_remote_file
    stat.st_ino, event.local_path, event.content_hash, stat.st_mtime
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/sync.py", line 939, in _save_local_hash
    self._db_manager_hash_cache.update(cache_entry)
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/utils/orm.py", line 559, in update
    if self.has(pk_sql):
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/utils/orm.py", line 510, in has
    result = self.db.execute(sql, pk_sql)
  File "/home/admin/.local/lib/python3.7/site-packages/maestral/utils/orm.py", line 290, in execute
    return self.connection.execute(sql, args)
OverflowError: Python int too large to convert to SQLite INTEGER
2022-02-15 15:02:56 manager INFO: Shutting down threads...
2022-02-15 15:02:56 sync INFO: Sync aborted
2022-02-15 15:02:56 manager INFO: Shutting down threads...
2022-02-15 15:02:56 sync INFO: Sync aborted
2022-02-15 15:02:56 manager INFO: Paused
```

> OS: Debian GNU/Linux 10 (buster) x86_64                        
> Kernel: 4.19.0-18-amd64                                        
> CPU: Intel Xeon (Skylake, IBRS) (2) @ 2.100GHz                 
> Memory: 1139MiB / 3850MiB

**Comment by samschott** [comment]
Damn, I did not think this could happen. The error occurs when trying to save the content hash of the downloaded file in the SQLite database. The content hash is saved together with the file path, modified time, and inode as form of caching to prevent duplicate computations. Of those three values, only the inode is saved in an `INTEGER` database table column.

AFAIK, SQLite allows up to 64-bit signed integers, though the issue might be that inodes are unsigned integers and may therefore be larger than `2^63 - 1`.

Which file system are using?

**Comment by samschott** [comment]
Could you also check the last file which it synced before one of those crashes with `maestral history`? What does `ls -li PATH` say about that file's inode number?

**Comment by jzebedee** [comment]
Will this fix be included in the next version? I currently can't sync anything with maestral because it will immediately crash with this error.

**Comment by samschott** [comment]
Yes, it will be. In the meantime, you can try out the dev release by installing:
```
pip3 install --pre maestral
```
It already includes the fix but no migration code for the database. It will therefore run a full reindexing on first run to rebuild our tables.

**Comment by jzebedee** [comment]
Thanks! I was able to complete a sync without errors after using `1.6.0.dev0`, but after some time it continues to crash with the same error:

```
$ maestral --version && maestral status && maestral log show | tail -n 30
1.6.0.dev0

Account:      xxxx
Usage:        108.4% of 17.3 GB used
Status:       Paused
Sync errors:  0


An unexpected error occurred
Please restart Maestral to continue syncing.

2022-03-03 12:40:10 manager INFO: Connecting...
2022-03-03 12:40:20 manager INFO: Connecting...
2022-03-03 22:52:37 manager INFO: Connected
2022-03-03 23:32:54 sync INFO: Fetching remote changes
2022-03-03 23:32:55 sync INFO: Indexing local changes...
2022-03-04 00:40:00 manager INFO: Connecting...
2022-03-04 00:40:11 manager INFO: Connected
2022-03-04 01:00:03 manager ERROR: Unexpected error
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/manager.py", line 808, in _handle_sync_thread_errors
    yield
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/manager.py", line 793, in startup_worker
    self.sync.upload_local_changes_while_inactive()
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/sync.py", line 1580, in upload_local_changes_while_inactive
    sync_events = [SyncEvent.from_file_system_event(e, self) for e in events]
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/sync.py", line 1580, in <listcomp>
    sync_events = [SyncEvent.from_file_system_event(e, self) for e in events]
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/models.py", line 399, in from_file_system_event
    content_hash = sync_engine.get_local_hash(to_path)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/sync.py", line 963, in get_local_hash
    cache_entry = self._hash_table.get(stat.st_ino)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/database/orm.py", line 345, in get
    result = self.db.execute(sql, pk_sql)
  File "/home/linuxbrew/.linuxbrew/opt/python@3.9/lib/python3.9/site-packages/maestral/database/orm.py", line 200, in execute
    return self.connection.execute(sql, args)
OverflowError: Python int too large to convert to SQLite INTEGER
2022-03-04 01:00:03 manager INFO: Shutting down threads...
2022-03-04 01:00:03 sync INFO: Sync aborted
2022-03-04 01:00:04 manager INFO: Paused
```

**Comment by samschott** [comment]
Hm, the pre-release version which contains the fix is `1.6.0.dev1`. It is possible that it wasn't uploaded to PyPI yet when I posted the message. Could you try to update again?
